### PR TITLE
House cleaning

### PR DESCRIPTION
**Pull request by pbrisbin**
- [x] Git ignore test.db
- [x] Stack lockfile
- [x] Circle config
- [x] Update to latest LTS
- [x] Restyle all files
- [x] Make CI happy

### CODE REVIEW

File: graphula-core/src/Graphula.hs [hunk 1]
```diff
@@ -617,7 +612,7 @@ attempt maxEdits maxInserts source = loop 0 0
 
 -- | For entities that only have singular 'Dependencies'. It uses data instead
 -- of newtype to match laziness of builtin tuples.
```
Comment (pbrisbin) [inline graphula-core/src/Graphula.hs:614]: Just noticed this comment. I guess I shouldn't blindly follow HLint? This laziness point is important?

Comment (cdparks) [reply graphula-core/src/Graphula.hs:614]: Eh, my opinion on this has evolved. Now I think a singleton tuple should be representationally identical to its parameter. Technically a breaking change, but no one should be relying on the indirection here. We can do a version bump once you change the exception types tho.

File: graphula-core/test/README.lhs [hunk 2]
```diff
@@ -18,6 +18,7 @@ Graphula is a simple interface for generating persistent data and linking its de
 {-# LANGUAGE ExistentialQuantification #-}
 {-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
```
Comment (cdparks) [inline graphula-core/test/README.lhs:21]: Persistent!

File: graphula-core/src/Graphula.hs (hunk 1 above)
Comment (pbrisbin) [reply graphula-core/src/Graphula.hs:614]: Cool, I'll nuke the comment before merging.

**Pull request merged**